li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -610,38 +623,84 @@ public void uncaughtException(Thread t, Throwable e) {
   }
 
   // Do synchronization on the method that will be closing the consumers since it will be handled by a different thread
-  synchronized void closeExistingConsumers(Map<String, String> newConfigs, UUID commandId) {
+  synchronized void recreateConsumers(Map<String, String> newConfigs, UUID commandId) {
     // TODO: ensureOpen() would throw an exception if consumers are already closed. In this case, we might want to send
     // an error message to notify mario that the config reload command fails due to consumer/producer already closed and
     // let mario decide what to do next (re-send the configs etc).
     ensureOpen();
 
+    Set<TopicPartition> curPartitions = assignment();
```

Comment:
This is an aggregate of assigned partitions. I think you should keep per-cluster assignment?